Draft: [messagingframework] SSO integration moved upstream.

dcaliste opened the pull request:
@pvuorela, this can be used to test the full compilation for device of the new developments to upstream authentication.

pvuorela commented:
The "post SSO" patch could perhaps have some more description on what are the remaining changes about. Also a lot of it is now USE_ACCOUNTS_QT so wondering should it be merged with the accounts-qt integration patch.

pvuorela commented:
This doesn't appear to apply on top of the qmf upstream PR (with both commits) now. Fails on "post SSO" patch:

error: patch failed: src/plugins/messageservices/smtp/smtpclient.cpp:629
error: src/plugins/messageservices/smtp/smtpclient.cpp: patch does not apply

That patch btw seems to also include qt style Change-Id. Couple existing ones also had it. No biggie but unnecessary here.

pvuorela commented on rpm/0002-Accounts-qt-integration.patch:
```diff
@@ -1932,6 +1946,304 @@ index 00000000..9f6ef320
++Manager* SSOAccountManager::_manager = 0;
++int SSOAccountManager::_refCount = 0;
++
++SSOAccountManager::SSOAccountManager()
```
The last usage for this we removed a while ago, right?

dcaliste replied:
Ah good point. I've reworked the QtAccounts integration patch not to create these files and use a direct Accounts::Manager object instead, like done in qmf-notification-plugin.

dcaliste commented:
I've adjusted the commit to take the latest changes from upstream and allow compilation. I've not tested it yet on device.

> That patch btw seems to also include qt style Change-Id.

I've removed them.

> The "post SSO" patch could perhaps have some more description

I've separated the post SSO patch into several logical patches:
- `0003-Use-a-specific-retry-delay-for-IMAP-idle-connections.patch`, this is not dependant on QtAccounts integration and could be proposed upstream,
- `0004-Don-t-save-settings-during-IMAP-logging-in.patch`, idem,
- `0028-Protect-service.patch`, this one looks very suspicious to me. I was not able to find when these changes were added and for what reason. I guess a crash coming from a memory corruption. If you look into `qmailmessageservice.cpp:370`, the `service` pointer is set on construction and is never change there after. For imap for instance, this is created in `imapservice.cpp:1545`. Except if a user of ImapService::source() is keeping a pointer on the return value, there is no reason for the source to out-live the servcie. Indeed in `servicehandler.cpp:946` and particularly in `ServiceHandler::registerAccountSource()`, the source pointer is saved in a map. So this could be problematic. But looking at the code, it seems that the map is flushed from the source before the service is deleted… To be investigated more. But the solution to protect the access to the service pointer as the patch is doing is useless in my opinion since the `_service` member can never be null by construction.
- `0029-Save-account-before-config.patch`, this change looks also very suspicious to me. The config of the account is changed on the line above. Calling a updateAccount(account) in a OR condition, will make that the config is not saved. I don't understand the purpose of the change.
- `0030-Don-t-fetch-capabilities-on-smtp-creation.patch`, this may go upstream somehow. It delays capabilities fetching when the credentials are not known on smtp service creation. I see the point of not starting an Internet connection on service creation, especially when capabilities require loging in and credentials are possibly not available yet. I need more thinking on how to do this in a clean way though… Maybe just removing the call like the patch is doing is the best way after all and delay capability fetching on first send…

dcaliste commented:
Included now in #20
